src/client-entry.ts

```diff
@@ -5,14 +5,16 @@ import TermboxRequest from '@/common/TermboxRequest';
 import { factory } from '@/common/TermboxFactory';
 import UlsLanguageTranslationRepository from '@/client/data-access/UlsLanguageTranslationRepository';
 import EntityRepository from '@/client/data-access/EntityRepository';
-import MwWindow from './client/MwWindow';
+import MwWindow from '@/client/mediawiki/MwWindow';
 
 Vue.config.productionTip = false;
 
 factory.setLanguageTranslationRepository(
 	new UlsLanguageTranslationRepository( ( window as MwWindow ).wb.getLanguageNameByCode ),
 );
-factory.setEntityRepository( new EntityRepository() );
+factory.setEntityRepository( new EntityRepository(
+	( window as MwWindow ).mw.hook( 'wikibase.entityPage.entityLoaded' ),
```

Comment:
Not that it makes any difference (+ rule n°3) but in an ideal world we would probably have a list of named constants of available hooks.

Reply:
Good idea. I added `src/client/mediawiki/Hooks.ts`